docs/step-by-step/step-05.md

```diff
@@ -0,0 +1,334 @@
+    crit: $this > (($status == $CRITICAL) ? (85) : (95))
+   delay: down 15m multiplier 1.5 max 1h
+    info: average cpu utilization for the last 10 minutes (excluding iowait, nice and steal)
+      to: sysadmin
+```
+
+To silence this alarm, change `sysadmin` to `silent`.
+
+```yaml
+      to: silent
+```
+
+Use `killall -USR2 netdata` to reload your health configuration. You can add `to: silence` to any alarm you'd rather not
+bother you with notifications.
+
+## Write your first health entity
+
+The best way to understand how health entities work is building your own and experimenting with the options. To start,
+let's build a health entity that triggers an alarm when system RAM usage goes above 80%.
+
+The first line in a health entity will be `alarm:`. This is how you name your entity. You can give it any name you
+choose, but the only symbols allowed are `.` and `_`. Let's call the alarm `ram_usage`.
+
+```yaml
+ alarm: ram_usage
+```
+
+> You'll see some funky indentation in the lines coming up. Don't worry about it too much! Indentation is not important
+> to how Netdata processes entities, and it will make sense when you're done.
+
+Next, you need to specify which chart this entity listens via the `on:` line. You're declaring that you want this alarm
+to check metrics on the `system.ram` chart.
+
+```yaml
+    on: system.ram
+```
+
+Now comes the `lookup`. This line specifies what metrics the alarm is looking for, what duration of time it's looking
+at, and how to process the metrics into a more usable format.
+
+```yaml
+lookup: average -1m percentage of used
+```
+
+Let's take a moment to break this line down.
+
+-   `average`: Calculate the average of all the metrics collected.
+-   `-1m`: Use metrics from 1 minute ago until now to calculate that average.
+-   `percentage`: Clarify that you want to calculate a percentage of RAM usage.
+-   `of used`: Specify which dimension (`used`) on the `system.ram` chart you want to monitor with this entity.
+
+In other words, you're taking 1 minute's worth of metrics from the `used` dimension on the `system.ram` chart,
+calculating their average, and returning it as a percentage.
+
+You can move on to the `units` line, which lets Netdata know that we're working with a percentage and not an absolute
+unit.
+
+```yaml
+ units: %
+```
+
+Next, the `every` line tells Netdata how often to perform the calculation you specified in the `lookup` line. For
+certain alarms, you might want to use a shorter duration, which you can specify using values like `10s`.
+
+```yaml
+ every: 1m
+```
+
+We'll put the next two lines—`warn` and `crit`—together. In these lines, you declare at which percentage you want to
+trigger a warning or critical alarm. Notice the variable `$this`, which is the value calculated by the `lookup` line.
+These lines will trigger a warning if that average RAM useag average goes above 80%, and a critical alert if it's above
+90%.
+
+```yaml
+  warn: $this > 80
+  crit: $this > 90
+```
```

Comment:
I would suggest adding something here at least mentioning the need for hysteresis to avoid alarm flapping. It's used by a vast majority of the stock alarms and the symptoms of not having it can be confusing to people who don't already have the background in mathematics or signal processing to know about it.

Reply:
I'll research hysteresis a bit more and maybe just add a small note about it. The last time I looked at the `crit: $this > (($status == $CRITICAL) ? (85) : (95))` I couldn't wrap my head around it in less than 2 minutes and so I moved on, but you're right that users should at least be introduced to the idea so they don't think "my" way is the best way to write entities forever.

Reply:
> I'll research hysteresis a bit more and maybe just add a small note about it. The last time I looked at the `crit: $this > (($status == $CRITICAL) ? (85) : (95))` I couldn't wrap my head around it in less than 2 minutes and so I moved on, but you're right that users should at least be introduced to the idea so they don't think "my" way is the best way to write entities forever.

Let me know if you need any help. I can't guarantee that I can make it perfectly clear, but a different perspective may help with understanding.

Reply:
> crit: $this > (($status == $CRITICAL) ? (85) : (95)) I couldn't wrap my head around it in less than 2 minutes and so I moved on

Haha completely understand, this is opposite of user friendly 😄

Reply:
> > crit: $this > (($status == $CRITICAL) ? (85) : (95)) I couldn't wrap my head around it in less than 2 minutes and so I moved on
> 
> Haha completely understand, this is opposite of user friendly

I mean, people who have a lot of background in C-like languages and a basic understanding of shell script/Perl shouldn't have any issues with the logic.

The problem is that other than the choice of symbology (and yes, I would much prefer a more verbose symbology), there really isn't a 'better' way to express hysteresis, you kind of need to show both trip points to make things unambiguous, and there's only so many ways you can do that without making the config parsing significantly more complex.

Reply:
I never realized there was already an explanation of hysteresis here: https://docs.netdata.cloud/health/reference/#special-use-of-the-conditional-operator. That's quite clear and helpful. So, I'll add a shorter version of this and a link to this section for them to learn more.

Reply:
... and done!